.github/workflows/ci-java7.yaml

```diff
@@ -0,0 +1,62 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Github action job to test core java library features on
+# downstream client libraries before they are released.
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+name: ci-java7
+jobs:
+  units:
+    name: "units (7)"
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-java@v1
+      # setup-java v2 or higher does not have version 1.7
+      with:
+        version: 1.7
+        architecture: x64
+    - run: |
+        java -version
+        # This value is used in "-Djvm=" later
+        echo "JAVA7_HOME=${JAVA_HOME}" >> $GITHUB_ENV
+    - uses: actions/setup-java@v3
+      with:
+        java-version: 17
+        distribution: temurin
+    - name: Set up Maven
+      uses: stCarolas/setup-maven@v4.5
+      with:
+        maven-version: 3.8.8
```

Comment:
I believe this is the last version of maven that supports Java 7, right? I don't think anyone is going to manually update this, but would it helpful to add a blurb as to why this set to `3.8.8` ?

Reply:
Actually this Maven does not need to work for Java 7. In fact this check does not run Maven via Java 7. (Maven calls Java 7 when running surefire unit tests)